mods/default/functions.lua

```diff
@@ -346,3 +346,40 @@ minetest.register_abm({
 	end
 })
 
+--
+-- Water mist 
+--
+
+minetest.register_abm({
+	nodenames = {"default:water_flowing"},
+	neighbors = {"default:water_source"},
+	interval = 5,
+	chance = 1,
+	action = function(pos, node, active_object_count, active_object_count_wider)
+		local posa = {x=pos.x, y=pos.y+1, z=pos.z}
+		local posu = {x=pos.x, y=pos.y-1, z=pos.z}
+		local na =  minetest.get_node(posa).name
+		local nu =  minetest.get_node(posu).name
+		if na == "default:water_flowing" then
+			if nu == "default:water_source" then
```

Comment:
This indentation gets very deep.  You can reduce it by two levels by combining and inverting the conditionals like so:
```Lua
if na != "default:water_flowing" or nu != "default:water_source" then
        return
end
minetest.add_particlespawner...
```